### PR TITLE
IBX-1410: Added `UserId`, `UserEmail` & `UserLogin` Criterions handling

### PR DESCRIPTION
**Pull request by Steveb-p**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-1410](https://issues.ibexa.co/browse/IBX-1410)
| **Type**                                   | improvement
| **Target Ibexa version** | `v3.3`
| **BC breaks**                          | no

This PR depends on:
 * https://github.com/ezsystems/ezplatform-kernel/pull/259

#### Checklist:

- [ ] Provided PR description.
- [x] Tested the solution manually.
- ~Provided automated test coverage.~
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`). 
- [x] Asked for a review (ping `@ibexa/engineering`).

### CODE REVIEW

File: src/lib/FieldMapper/ContentFieldMapper/UserDocumentFields.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            return [];
+        }
+
+        $fields = [];
+
+        if (isset($userField->value->externalData['login'])) {
+            $fields[] = new Field(
+                'user_login',
+                hash(self::HASHING_ALGORITHM, $userField->value->externalData['login']),
+                new FieldType\StringField()
```
Comment (adamwojs) [inline src/lib/FieldMapper/ContentFieldMapper/UserDocumentFields.php:39]: Actually `IdentifierField` seem to be better choice then `StringField`

File: src/lib/Query/Common/CriterionVisitor/UserEmailIn.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+final class UserEmailIn extends CriterionVisitor
+{
+    public function canVisit(Criterion $criterion): bool
+    {
+        $operator = $criterion->operator ?? Operator::IN;
+        $supportedOperators = [Operator::IN, Operator::EQ];
+
+        return
+            $criterion instanceof Criterion\UserEmail
+            && in_array($operator, $supportedOperators, true);
```
Comment (adamwojs) [inline src/lib/Query/Common/CriterionVisitor/UserEmailIn.php:25]:
```suggestion
        if ($criterion instanceof Criterion\UserEmail) {
            return in_array($operator, $supportedOperators, true);
        }
        
        return false;    
```

File: src/lib/Query/Common/CriterionVisitor/UserEmailIn.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\Operator;
+use EzSystems\EzPlatformSolrSearchEngine\Query\CriterionVisitor;
+use Ibexa\Solr\FieldMapper\ContentFieldMapper\UserDocumentFields;
+
+final class UserEmailIn extends CriterionVisitor
+{
+    public function canVisit(Criterion $criterion): bool
+    {
+        $operator = $criterion->operator ?? Operator::IN;
+        $supportedOperators = [Operator::IN, Operator::EQ];
```
Comment (adamwojs) [inline src/lib/Query/Common/CriterionVisitor/UserEmailIn.php:21]: You can declare supported operators as `private const` or just inline value if you decide to apply next suggestion

File: src/lib/FieldMapper/ContentFieldMapper/UserDocumentFields.php (hunk 1 above)
Comment (Steveb-p) [reply src/lib/FieldMapper/ContentFieldMapper/UserDocumentFields.php]: @adamwojs changed to `IdentifierField` in 048cdbf4.

File: src/lib/Query/Common/CriterionVisitor/UserEmailIn.php (hunk 2 above)
Comment (Steveb-p) [reply src/lib/Query/Common/CriterionVisitor/UserEmailIn.php]: Inlined variables in 75f609d2.

File: src/lib/Query/Common/CriterionVisitor/UserEmailIn.php (hunk 3 above)
Comment (Steveb-p) [reply src/lib/Query/Common/CriterionVisitor/UserEmailIn.php]: Inlined variables in 75f609d2.

File: src/lib/Query/Common/CriterionVisitor/UserEmailIn.php (hunk 2 above)
Comment (Steveb-p) [reply src/lib/Query/Common/CriterionVisitor/UserEmailIn.php]: Changed implementation to:
```php
    public function canVisit(Criterion $criterion): bool
    {
        if (!$criterion instanceof Criterion\UserLogin) {
            return false;
        }

        return in_array($criterion->operator ?? Operator::IN, [Operator::IN, Operator::EQ], true);
    }
```

**Review by adamwojs — APPROVED** [review]

File: composer.json [hunk 4]
```diff
@@ -29,9 +29,11 @@
...
     "autoload": {
         "psr-4": {
-            "EzSystems\\EzPlatformSolrSearchEngine\\": "lib",
-            "EzSystems\\EzPlatformSolrSearchEngineBundle\\": "bundle",
-            "EzSystems\\EzPlatformSolrSearchEngine\\Tests\\SetupFactory\\": "tests/lib/SetupFactory"
+            "Ibexa\\Solr\\": "src/lib/",
+            "EzSystems\\EzPlatformSolrSearchEngine\\": "lib/",
+            "EzSystems\\EzPlatformSolrSearchEngineBundle\\": "bundle/",
+            "EzSystems\\EzPlatformSolrSearchEngine\\Tests\\SetupFactory\\": "tests/lib/SetupFactory/",
+            "Ibexa\\Tests\\Solr\\SetupFactory\\": "tests/lib/SetupFactory/"
```
Comment (alongosz) [inline composer.json:36]: probably should be above `EzSystems` as well.

**Review by alongosz — APPROVED** [review]

File: composer.json (hunk 4 above)
Comment (Steveb-p) [reply composer.json]: Fixed in 9c9e878a.

**Comment by Steveb-p** [comment]
Merged up into `master` in c8edc829.

**Pull request merged**